[firetv] [Regression] @flexn/template can't be built on firetv on new rnv project

rnv version: 0.36.1-canary.2
flexn template version: 0.16.0

How to repro:

1. Create new project using `rnv new` and select @flexn/template
2. Do `rnv build -p firetv` (same happens when running)

```
┌──────────────────────────────────────────────────────────────────────────────┐
│  🚀  SUMMARY | 6/22/2022, 3:59:09 PM                                         │
│  $ rnv build -p firetv                                                       │
├──────────────────────────────────────────────────────────────────────────────┤
│  ReNative Version: 0.36.1-canary.2                                           │
│  Project Name ($package.name): flexnTemplate16                               │
│  Project Version ($package.version): 0.1.0                                   │
│  Workspace ($.workspaceID): flexn                                            │
│  Platform (-p): firetv                                                       │
│  Engine ($.platforms.firetv.engine): engine-rn-tvos                          │
│  Template: @flexn/template@0.16.0                                            │
│  App Config (-c): app                                                        │
│  Build Scheme (-s): debug                                                    │
│  Target (-t): Android_TV_720p_API_22                                         │
│  Supported Platforms: firetv                                                 │
│  Env Info: darwin | x64 | node v16.14.0                                      │
│  Executed Time: 0h:0m:3s:744ms                                               │
│                                                                              │
│ [ error ] [build] COMMAND:

./gradlew assembleDebug -x bundleReleaseJsAndAssets

FAILED with ERROR:

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s

Command failed with exit code 1: ./gradlew assembleDebug -x bundleReleaseJsAndAssets
e: /Users/pauliusguzas/Desktop/flexnTemplate16/platformBuilds/app_firetv/app/src/main/java/rnv/MainActivity.kt: (39, 64): No value passed for parameter 'type'

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Configuration on demand is an incubating feature.

> Configure project :app
WARNING:: The option 'android.useDeprecatedNdk' is deprecated.
The current default is 'false'.
It has been removed from the current version of the Android Gradle plugin.
NdkCompile is no longer supported
Warning: The 'kotlin-android-extensions' Gradle plugin is deprecated. Please use this migration guide (https://goo.gle/kotlin-android-extensions-deprecation) to start working with View Binding (https://developer.android.com/topic/libraries/view-binding) and the 'kotlin-parcelize' plugin.

> Configure project :react-native-reanimated
WARNING:: The specified Android SDK Build Tools version (28.0.0) is ignored, as it is below the minimum supported version (30.0.2) for Android Gradle Plugin 4.2.2.
Android SDK Build Tools 30.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.0'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.

> Task :react-native-vector-icons:preBuild UP-TO-DATE
> Task :react-native-gesture-handler:preBuild UP-TO-DATE
> Task :react-native-screens:preBuild UP-TO-DATE
> Task :react-native-safe-area-context:preBuild UP-TO-DATE
> Task :flexn-io-sdk:preBuild UP-TO-DATE
> Task :react-native-reanimated:preBuild UP-TO-DATE
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :react-native-gesture-handler:preDebugBuild UP-TO-DATE
> Task :flexn-io-sdk:preDebugBuild UP-TO-DATE
> Task :react-native-reanimated:preDebugBuild UP-TO-DATE
> Task :react-native-vector-icons:preDebugBuild UP-TO-DATE
> Task :react-native-screens:preDebugBuild UP-TO-DATE
> Task :react-native-safe-area-context:preDebugBuild UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:compileDebugAidl NO-SOURCE
> Task :react-native-reanimated:compileDebugAidl NO-SOURCE
> Task :react-native-vector-icons:compileDebugAidl NO-SOURCE
> Task :react-native-screens:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:packageDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:packageDebugRenderscript NO-SOURCE
> Task :flexn-io-sdk:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:writeDebugAarMetadata UP-TO-DATE
> Task :app:generateDebugBuildConfig UP-TO-DATE
> Task :app:bundleDebugJsAndAssets SKIPPED
> Task :app:mergeDebugShaders UP-TO-DATE
> Task :app:compileDebugShaders NO-SOURCE
> Task :app:generateDebugAssets UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugShaders UP-TO-DATE
> Task :app:generateDebugResValues UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugShaders NO-SOURCE
> Task :react-native-safe-area-context:generateDebugAssets UP-TO-DATE
> Task :flexn-io-sdk:packageDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:mergeDebugShaders UP-TO-DATE
> Task :flexn-io-sdk:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:packageDebugAssets UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugShaders UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugShaders NO-SOURCE
> Task :react-native-screens:packageDebugRenderscript NO-SOURCE
> Task :react-native-screens:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugAssets UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugRenderscript NO-SOURCE
> Task :react-native-vector-icons:packageDebugRenderscript NO-SOURCE
> Task :app:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:packageDebugAssets UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugResValues UP-TO-DATE
> Task :react-native-screens:mergeDebugShaders UP-TO-DATE
> Task :react-native-vector-icons:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugResources UP-TO-DATE
> Task :react-native-reanimated:packageDebugRenderscript NO-SOURCE
> Task :flexn-io-sdk:compileDebugShaders NO-SOURCE
> Task :react-native-reanimated:writeDebugAarMetadata UP-TO-DATE
> Task :flexn-io-sdk:generateDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugRenderscript NO-SOURCE
> Task :flexn-io-sdk:packageDebugAssets UP-TO-DATE
> Task :react-native-reanimated:mergeDebugShaders UP-TO-DATE
> Task :react-native-screens:compileDebugShaders NO-SOURCE
> Task :react-native-vector-icons:mergeDebugShaders UP-TO-DATE
> Task :react-native-screens:generateDebugAssets UP-TO-DATE
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:generateDebugResValues UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugResources UP-TO-DATE
> Task :react-native-screens:packageDebugAssets UP-TO-DATE
> Task :react-native-reanimated:compileDebugShaders NO-SOURCE
> Task :react-native-reanimated:generateDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:packageDebugResources UP-TO-DATE
> Task :react-native-reanimated:packageDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:extractDeepLinksDebug UP-TO-DATE
> Task :flexn-io-sdk:compileDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:packageDebugResources UP-TO-DATE
> Task :flexn-io-sdk:generateDebugResValues UP-TO-DATE
> Task :flexn-io-sdk:generateDebugResources UP-TO-DATE
> Task :react-native-safe-area-context:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-vector-icons:compileDebugShaders NO-SOURCE
> Task :react-native-vector-icons:generateDebugAssets UP-TO-DATE
> Task :react-native-vector-icons:packageDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:processDebugManifest UP-TO-DATE
> Task :react-native-screens:compileDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:processDebugManifest UP-TO-DATE
> Task :flexn-io-sdk:packageDebugResources UP-TO-DATE
> Task :react-native-reanimated:compileDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-vector-icons:compileDebugRenderscript NO-SOURCE
> Task :react-native-vector-icons:generateDebugResValues UP-TO-DATE
> Task :flexn-io-sdk:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-reanimated:generateDebugResValues UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-screens:generateDebugResValues UP-TO-DATE
> Task :react-native-reanimated:generateDebugResources UP-TO-DATE
> Task :react-native-screens:generateDebugResources UP-TO-DATE
> Task :react-native-vector-icons:generateDebugResources UP-TO-DATE
> Task :react-native-gesture-handler:parseDebugLocalResources UP-TO-DATE
> Task :react-native-reanimated:packageDebugResources UP-TO-DATE
> Task :react-native-reanimated:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-vector-icons:packageDebugResources UP-TO-DATE
> Task :flexn-io-sdk:processDebugManifest UP-TO-DATE
> Task :react-native-vector-icons:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-safe-area-context:parseDebugLocalResources UP-TO-DATE
> Task :react-native-screens:packageDebugResources UP-TO-DATE
> Task :react-native-screens:extractDeepLinksDebug UP-TO-DATE
> Task :flexn-io-sdk:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-reanimated:processDebugManifest UP-TO-DATE
> Task :react-native-reanimated:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-vector-icons:processDebugManifest UP-TO-DATE
> Task :react-native-vector-icons:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-screens:processDebugManifest UP-TO-DATE
> Task :react-native-screens:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-reanimated:parseDebugLocalResources UP-TO-DATE
> Task :react-native-vector-icons:parseDebugLocalResources UP-TO-DATE
> Task :react-native-screens:parseDebugLocalResources UP-TO-DATE
> Task :flexn-io-sdk:parseDebugLocalResources UP-TO-DATE
> Task :react-native-reanimated:generateDebugRFile UP-TO-DATE
> Task :flexn-io-sdk:generateDebugRFile UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugRFile UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugRFile UP-TO-DATE
> Task :react-native-vector-icons:generateDebugRFile UP-TO-DATE
> Task :flexn-io-sdk:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-reanimated:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-vector-icons:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-safe-area-context:javaPreCompileDebug UP-TO-DATE
> Task :react-native-gesture-handler:javaPreCompileDebug UP-TO-DATE
> Task :flexn-io-sdk:javaPreCompileDebug UP-TO-DATE
> Task :react-native-reanimated:javaPreCompileDebug UP-TO-DATE
> Task :react-native-vector-icons:javaPreCompileDebug UP-TO-DATE
> Task :app:checkDebugAarMetadata UP-TO-DATE
> Task :app:mergeDebugAssets UP-TO-DATE
> Task :app:copyDebugBundledJs SKIPPED
> Task :app:generateDebugResources UP-TO-DATE
> Task :react-native-screens:generateDebugRFile UP-TO-DATE
> Task :react-native-screens:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-screens:javaPreCompileDebug UP-TO-DATE
> Task :react-native-screens:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-gesture-handler:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-screens:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-screens:processDebugJavaRes NO-SOURCE
> Task :react-native-gesture-handler:processDebugJavaRes NO-SOURCE
> Task :react-native-screens:bundleLibResDebug NO-SOURCE
> Task :react-native-gesture-handler:bundleLibResDebug NO-SOURCE
> Task :react-native-screens:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-screens:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-gesture-handler:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-screens:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-screens:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-screens:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-screens:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-gesture-handler:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-gesture-handler:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-gesture-handler:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-reanimated:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-reanimated:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-reanimated:processDebugJavaRes NO-SOURCE
> Task :react-native-vector-icons:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-reanimated:bundleLibResDebug NO-SOURCE
> Task :react-native-reanimated:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-reanimated:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-reanimated:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-reanimated:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-reanimated:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-reanimated:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :app:mergeDebugResources UP-TO-DATE
> Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
> Task :app:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-vector-icons:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-safe-area-context:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-safe-area-context:processDebugJavaRes NO-SOURCE
> Task :react-native-safe-area-context:bundleLibResDebug NO-SOURCE
> Task :react-native-safe-area-context:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-safe-area-context:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-safe-area-context:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-safe-area-context:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-vector-icons:processDebugJavaRes NO-SOURCE
> Task :react-native-vector-icons:bundleLibResDebug NO-SOURCE
> Task :react-native-vector-icons:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-screens:extractDebugAnnotations UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-vector-icons:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-vector-icons:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-screens:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-screens:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :react-native-vector-icons:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-screens:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-reanimated:extractDebugAnnotations UP-TO-DATE
> Task :react-native-screens:mergeDebugJavaResource UP-TO-DATE
> Task :react-native-screens:syncDebugLibJars UP-TO-DATE
> Task :react-native-reanimated:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-screens:bundleDebugAar UP-TO-DATE
> Task :react-native-screens:compileDebugSources UP-TO-DATE
> Task :react-native-screens:assembleDebug UP-TO-DATE
> Task :react-native-reanimated:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :react-native-vector-icons:extractDebugAnnotations UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-vector-icons:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-reanimated:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-reanimated:mergeDebugJavaResource UP-TO-DATE
> Task :react-native-safe-area-context:extractDebugAnnotations UP-TO-DATE
> Task :react-native-gesture-handler:extractDebugAnnotations UP-TO-DATE
> Task :app:processDebugMainManifest UP-TO-DATE
> Task :flexn-io-sdk:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-reanimated:syncDebugLibJars UP-TO-DATE
> Task :app:processDebugManifest UP-TO-DATE
> Task :react-native-reanimated:bundleDebugAar UP-TO-DATE
> Task :app:processDebugManifestForPackage UP-TO-DATE
> Task :react-native-reanimated:compileDebugSources UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:processDebugJavaRes NO-SOURCE
> Task :react-native-safe-area-context:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:bundleLibResDebug NO-SOURCE
> Task :react-native-reanimated:assembleDebug UP-TO-DATE
> Task :flexn-io-sdk:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-safe-area-context:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugJavaResource UP-TO-DATE
> Task :react-native-gesture-handler:prepareLintJarForPublish UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-vector-icons:mergeDebugJavaResource UP-TO-DATE
> Task :flexn-io-sdk:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-safe-area-context:syncDebugLibJars UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugJavaResource UP-TO-DATE
> Task :flexn-io-sdk:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :flexn-io-sdk:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-gesture-handler:syncDebugLibJars UP-TO-DATE
> Task :react-native-safe-area-context:bundleDebugAar UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugSources UP-TO-DATE
> Task :react-native-safe-area-context:assembleDebug UP-TO-DATE
> Task :react-native-gesture-handler:bundleDebugAar UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugSources UP-TO-DATE
> Task :react-native-gesture-handler:assembleDebug UP-TO-DATE
> Task :react-native-vector-icons:syncDebugLibJars UP-TO-DATE
> Task :react-native-vector-icons:bundleDebugAar UP-TO-DATE
> Task :react-native-vector-icons:compileDebugSources UP-TO-DATE
> Task :react-native-vector-icons:assembleDebug UP-TO-DATE
> Task :flexn-io-sdk:extractDebugAnnotations UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:prepareLintJarForPublish UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugJavaResource UP-TO-DATE
> Task :flexn-io-sdk:syncDebugLibJars UP-TO-DATE
> Task :flexn-io-sdk:bundleDebugAar UP-TO-DATE
> Task :flexn-io-sdk:compileDebugSources UP-TO-DATE
> Task :flexn-io-sdk:assembleDebug UP-TO-DATE
> Task :app:processDebugResources UP-TO-DATE

> Task :app:compileDebugKotlin FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9.1/userguide/command_line_interface.html#sec:command_line_warnings
163 actionable tasks: 1 executed, 162 up-to-date

e: /Users/pauliusguzas/Desktop/flexnTemplate16/platformBuilds/app_firetv/app/src/main/java/rnv/MainActivity.kt: (39, 64): No value passed for parameter 'type'

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s

Error: Command failed with exit code 1: ./gradlew assembleDebug -x bundleReleaseJsAndAssets
e: /Users/pauliusguzas/Desktop/flexnTemplate16/platformBuilds/app_firetv/app/src/main/java/rnv/MainActivity.kt: (39, 64): No value passed for parameter 'type'

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Configuration on demand is an incubating feature.

> Configure project :app
WARNING:: The option 'android.useDeprecatedNdk' is deprecated.
The current default is 'false'.
It has been removed from the current version of the Android Gradle plugin.
NdkCompile is no longer supported
Warning: The 'kotlin-android-extensions' Gradle plugin is deprecated. Please use this migration guide (https://goo.gle/kotlin-android-extensions-deprecation) to start working with View Binding (https://developer.android.com/topic/libraries/view-binding) and the 'kotlin-parcelize' plugin.

> Configure project :react-native-reanimated
WARNING:: The specified Android SDK Build Tools version (28.0.0) is ignored, as it is below the minimum supported version (30.0.2) for Android Gradle Plugin 4.2.2.
Android SDK Build Tools 30.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.0'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.

> Task :react-native-vector-icons:preBuild UP-TO-DATE
> Task :react-native-gesture-handler:preBuild UP-TO-DATE
> Task :react-native-screens:preBuild UP-TO-DATE
> Task :react-native-safe-area-context:preBuild UP-TO-DATE
> Task :flexn-io-sdk:preBuild UP-TO-DATE
> Task :react-native-reanimated:preBuild UP-TO-DATE
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :react-native-gesture-handler:preDebugBuild UP-TO-DATE
> Task :flexn-io-sdk:preDebugBuild UP-TO-DATE
> Task :react-native-reanimated:preDebugBuild UP-TO-DATE
> Task :react-native-vector-icons:preDebugBuild UP-TO-DATE
> Task :react-native-screens:preDebugBuild UP-TO-DATE
> Task :react-native-safe-area-context:preDebugBuild UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:compileDebugAidl NO-SOURCE
> Task :react-native-reanimated:compileDebugAidl NO-SOURCE
> Task :react-native-vector-icons:compileDebugAidl NO-SOURCE
> Task :react-native-screens:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:packageDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:packageDebugRenderscript NO-SOURCE
> Task :flexn-io-sdk:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:writeDebugAarMetadata UP-TO-DATE
> Task :app:generateDebugBuildConfig UP-TO-DATE
> Task :app:bundleDebugJsAndAssets SKIPPED
> Task :app:mergeDebugShaders UP-TO-DATE
> Task :app:compileDebugShaders NO-SOURCE
> Task :app:generateDebugAssets UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugShaders UP-TO-DATE
> Task :app:generateDebugResValues UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugShaders NO-SOURCE
> Task :react-native-safe-area-context:generateDebugAssets UP-TO-DATE
> Task :flexn-io-sdk:packageDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:mergeDebugShaders UP-TO-DATE
> Task :flexn-io-sdk:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:packageDebugAssets UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugShaders UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugShaders NO-SOURCE
> Task :react-native-screens:packageDebugRenderscript NO-SOURCE
> Task :react-native-screens:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugAssets UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugRenderscript NO-SOURCE
> Task :react-native-vector-icons:packageDebugRenderscript NO-SOURCE
> Task :app:compileDebugAidl NO-SOURCE
> Task :react-native-gesture-handler:packageDebugAssets UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugResValues UP-TO-DATE
> Task :react-native-screens:mergeDebugShaders UP-TO-DATE
> Task :react-native-vector-icons:writeDebugAarMetadata UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugResources UP-TO-DATE
> Task :react-native-reanimated:packageDebugRenderscript NO-SOURCE
> Task :flexn-io-sdk:compileDebugShaders NO-SOURCE
> Task :react-native-reanimated:writeDebugAarMetadata UP-TO-DATE
> Task :flexn-io-sdk:generateDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugRenderscript NO-SOURCE
> Task :flexn-io-sdk:packageDebugAssets UP-TO-DATE
> Task :react-native-reanimated:mergeDebugShaders UP-TO-DATE
> Task :react-native-screens:compileDebugShaders NO-SOURCE
> Task :react-native-vector-icons:mergeDebugShaders UP-TO-DATE
> Task :react-native-screens:generateDebugAssets UP-TO-DATE
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:generateDebugResValues UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugResources UP-TO-DATE
> Task :react-native-screens:packageDebugAssets UP-TO-DATE
> Task :react-native-reanimated:compileDebugShaders NO-SOURCE
> Task :react-native-reanimated:generateDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:packageDebugResources UP-TO-DATE
> Task :react-native-reanimated:packageDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:extractDeepLinksDebug UP-TO-DATE
> Task :flexn-io-sdk:compileDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:packageDebugResources UP-TO-DATE
> Task :flexn-io-sdk:generateDebugResValues UP-TO-DATE
> Task :flexn-io-sdk:generateDebugResources UP-TO-DATE
> Task :react-native-safe-area-context:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-vector-icons:compileDebugShaders NO-SOURCE
> Task :react-native-vector-icons:generateDebugAssets UP-TO-DATE
> Task :react-native-vector-icons:packageDebugAssets UP-TO-DATE
> Task :react-native-gesture-handler:processDebugManifest UP-TO-DATE
> Task :react-native-screens:compileDebugRenderscript NO-SOURCE
> Task :react-native-safe-area-context:processDebugManifest UP-TO-DATE
> Task :flexn-io-sdk:packageDebugResources UP-TO-DATE
> Task :react-native-reanimated:compileDebugRenderscript NO-SOURCE
> Task :react-native-gesture-handler:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-vector-icons:compileDebugRenderscript NO-SOURCE
> Task :react-native-vector-icons:generateDebugResValues UP-TO-DATE
> Task :flexn-io-sdk:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-reanimated:generateDebugResValues UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-screens:generateDebugResValues UP-TO-DATE
> Task :react-native-reanimated:generateDebugResources UP-TO-DATE
> Task :react-native-screens:generateDebugResources UP-TO-DATE
> Task :react-native-vector-icons:generateDebugResources UP-TO-DATE
> Task :react-native-gesture-handler:parseDebugLocalResources UP-TO-DATE
> Task :react-native-reanimated:packageDebugResources UP-TO-DATE
> Task :react-native-reanimated:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-vector-icons:packageDebugResources UP-TO-DATE
> Task :flexn-io-sdk:processDebugManifest UP-TO-DATE
> Task :react-native-vector-icons:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-safe-area-context:parseDebugLocalResources UP-TO-DATE
> Task :react-native-screens:packageDebugResources UP-TO-DATE
> Task :react-native-screens:extractDeepLinksDebug UP-TO-DATE
> Task :flexn-io-sdk:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-reanimated:processDebugManifest UP-TO-DATE
> Task :react-native-reanimated:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-vector-icons:processDebugManifest UP-TO-DATE
> Task :react-native-vector-icons:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-screens:processDebugManifest UP-TO-DATE
> Task :react-native-screens:compileDebugLibraryResources UP-TO-DATE
> Task :react-native-reanimated:parseDebugLocalResources UP-TO-DATE
> Task :react-native-vector-icons:parseDebugLocalResources UP-TO-DATE
> Task :react-native-screens:parseDebugLocalResources UP-TO-DATE
> Task :flexn-io-sdk:parseDebugLocalResources UP-TO-DATE
> Task :react-native-reanimated:generateDebugRFile UP-TO-DATE
> Task :flexn-io-sdk:generateDebugRFile UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugRFile UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugRFile UP-TO-DATE
> Task :react-native-vector-icons:generateDebugRFile UP-TO-DATE
> Task :flexn-io-sdk:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-gesture-handler:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-safe-area-context:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-reanimated:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-vector-icons:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-safe-area-context:javaPreCompileDebug UP-TO-DATE
> Task :react-native-gesture-handler:javaPreCompileDebug UP-TO-DATE
> Task :flexn-io-sdk:javaPreCompileDebug UP-TO-DATE
> Task :react-native-reanimated:javaPreCompileDebug UP-TO-DATE
> Task :react-native-vector-icons:javaPreCompileDebug UP-TO-DATE
> Task :app:checkDebugAarMetadata UP-TO-DATE
> Task :app:mergeDebugAssets UP-TO-DATE
> Task :app:copyDebugBundledJs SKIPPED
> Task :app:generateDebugResources UP-TO-DATE
> Task :react-native-screens:generateDebugRFile UP-TO-DATE
> Task :react-native-screens:generateDebugBuildConfig UP-TO-DATE
> Task :react-native-screens:javaPreCompileDebug UP-TO-DATE
> Task :react-native-screens:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-gesture-handler:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-screens:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-screens:processDebugJavaRes NO-SOURCE
> Task :react-native-gesture-handler:processDebugJavaRes NO-SOURCE
> Task :react-native-screens:bundleLibResDebug NO-SOURCE
> Task :react-native-gesture-handler:bundleLibResDebug NO-SOURCE
> Task :react-native-screens:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-screens:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-gesture-handler:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-screens:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-screens:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-screens:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-screens:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-gesture-handler:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-gesture-handler:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-gesture-handler:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-reanimated:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-reanimated:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-reanimated:processDebugJavaRes NO-SOURCE
> Task :react-native-vector-icons:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-reanimated:bundleLibResDebug NO-SOURCE
> Task :react-native-reanimated:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-reanimated:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-reanimated:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-reanimated:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-reanimated:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-reanimated:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :app:mergeDebugResources UP-TO-DATE
> Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
> Task :app:extractDeepLinksDebug UP-TO-DATE
> Task :react-native-vector-icons:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-safe-area-context:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-safe-area-context:processDebugJavaRes NO-SOURCE
> Task :react-native-safe-area-context:bundleLibResDebug NO-SOURCE
> Task :react-native-safe-area-context:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-safe-area-context:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-safe-area-context:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-safe-area-context:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-vector-icons:processDebugJavaRes NO-SOURCE
> Task :react-native-vector-icons:bundleLibResDebug NO-SOURCE
> Task :react-native-vector-icons:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-screens:extractDebugAnnotations UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-vector-icons:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-vector-icons:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :react-native-screens:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-screens:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :react-native-vector-icons:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-screens:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-reanimated:extractDebugAnnotations UP-TO-DATE
> Task :react-native-screens:mergeDebugJavaResource UP-TO-DATE
> Task :react-native-screens:syncDebugLibJars UP-TO-DATE
> Task :react-native-reanimated:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-screens:bundleDebugAar UP-TO-DATE
> Task :react-native-screens:compileDebugSources UP-TO-DATE
> Task :react-native-screens:assembleDebug UP-TO-DATE
> Task :react-native-reanimated:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :react-native-vector-icons:extractDebugAnnotations UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-vector-icons:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:compileDebugJavaWithJavac UP-TO-DATE
> Task :react-native-vector-icons:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-reanimated:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-reanimated:mergeDebugJavaResource UP-TO-DATE
> Task :react-native-safe-area-context:extractDebugAnnotations UP-TO-DATE
> Task :react-native-gesture-handler:extractDebugAnnotations UP-TO-DATE
> Task :app:processDebugMainManifest UP-TO-DATE
> Task :flexn-io-sdk:bundleLibCompileToJarDebug UP-TO-DATE
> Task :react-native-reanimated:syncDebugLibJars UP-TO-DATE
> Task :app:processDebugManifest UP-TO-DATE
> Task :react-native-reanimated:bundleDebugAar UP-TO-DATE
> Task :app:processDebugManifestForPackage UP-TO-DATE
> Task :react-native-reanimated:compileDebugSources UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:processDebugJavaRes NO-SOURCE
> Task :react-native-safe-area-context:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:bundleLibResDebug NO-SOURCE
> Task :react-native-reanimated:assembleDebug UP-TO-DATE
> Task :flexn-io-sdk:bundleLibRuntimeToJarDebug UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :react-native-safe-area-context:prepareLintJarForPublish UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugJniLibFolders UP-TO-DATE
> Task :react-native-safe-area-context:mergeDebugJavaResource UP-TO-DATE
> Task :react-native-gesture-handler:prepareLintJarForPublish UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugNativeLibs NO-SOURCE
> Task :react-native-vector-icons:mergeDebugJavaResource UP-TO-DATE
> Task :flexn-io-sdk:stripDebugDebugSymbols NO-SOURCE
> Task :react-native-safe-area-context:syncDebugLibJars UP-TO-DATE
> Task :react-native-gesture-handler:mergeDebugJavaResource UP-TO-DATE
> Task :flexn-io-sdk:copyDebugJniLibsProjectOnly UP-TO-DATE
> Task :flexn-io-sdk:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
> Task :react-native-gesture-handler:syncDebugLibJars UP-TO-DATE
> Task :react-native-safe-area-context:bundleDebugAar UP-TO-DATE
> Task :react-native-safe-area-context:compileDebugSources UP-TO-DATE
> Task :react-native-safe-area-context:assembleDebug UP-TO-DATE
> Task :react-native-gesture-handler:bundleDebugAar UP-TO-DATE
> Task :react-native-gesture-handler:compileDebugSources UP-TO-DATE
> Task :react-native-gesture-handler:assembleDebug UP-TO-DATE
> Task :react-native-vector-icons:syncDebugLibJars UP-TO-DATE
> Task :react-native-vector-icons:bundleDebugAar UP-TO-DATE
> Task :react-native-vector-icons:compileDebugSources UP-TO-DATE
> Task :react-native-vector-icons:assembleDebug UP-TO-DATE
> Task :flexn-io-sdk:extractDebugAnnotations UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugGeneratedProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugConsumerProguardFiles UP-TO-DATE
> Task :flexn-io-sdk:prepareLintJarForPublish UP-TO-DATE
> Task :flexn-io-sdk:mergeDebugJavaResource UP-TO-DATE
> Task :flexn-io-sdk:syncDebugLibJars UP-TO-DATE
> Task :flexn-io-sdk:bundleDebugAar UP-TO-DATE
> Task :flexn-io-sdk:compileDebugSources UP-TO-DATE
> Task :flexn-io-sdk:assembleDebug UP-TO-DATE
> Task :app:processDebugResources UP-TO-DATE

> Task :app:compileDebugKotlin FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9.1/userguide/command_line_interface.html#sec:command_line_warnings
163 actionable tasks: 1 executed, 162 up-to-date
    at makeError (/Users/pauliusguzas/Desktop/flexnTemplate16/node_modules/rnv/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/Users/pauliusguzas/Desktop/flexnTemplate16/node_modules/rnv/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

--- a/packages/harness/renative.json
+++ b/packages/harness/renative.json
@@ -84,7 +84,25 @@
                     "io.flexn.sdk.TvRemoteHandlerModule",
                     "android.view.KeyEvent;"
                 ],
-                "activityMethods": []
+                "activityMethods": [
+                    "override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {",
+                    "TvRemoteHandlerModule.getInstance().onKeyEvent(event, \"up\");",
+                    "return super.onKeyUp(keyCode, event)",
+                    "}",
+                    "override fun onKeyLongPress(keyCode: Int, event: KeyEvent?): Boolean {",
+                    "   TvRemoteHandlerModule.getInstance().onKeyEvent(event,  \"longPress\");",
+                    "    return super.onKeyLongPress(keyCode, event)",
+                    "}",
+                    "override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {",
+                    "if(keyCode == KeyEvent.KEYCODE_DPAD_RIGHT || keyCode == KeyEvent.KEYCODE_DPAD_LEFT || keyCode == KeyEvent.KEYCODE_DPAD_UP || keyCode == KeyEvent.KEYCODE_DPAD_DOWN) {",
+                    "event?.startTracking();",
+                    "TvRemoteHandlerModule.getInstance().onKeyEvent(event,\"down\");",
+                    "return true;",
+                    "}",
+                    "TvRemoteHandlerModule.getInstance().onKeyEvent(event, \"down\");",
+                    "return super.onKeyDown(keyCode, event)",
+                    "}"
+                ]
             }
         },
         "@rnv/renative": "source:flexn",

--- a/packages/template/renative.json
+++ b/packages/template/renative.json
@@ -98,11 +98,22 @@
                     "android.view.KeyEvent;"
                 ],
                 "activityMethods": [
-                    "override fun dispatchKeyEvent(event: KeyEvent): Boolean {",
-                    "   if (event.action == KeyEvent.ACTION_DOWN || event.action == KeyEvent.ACTION_UP) {",
-                    "       TvRemoteHandlerModule.getInstance().onKeyEvent(event);",
-                    "   }",
-                    "   return super.dispatchKeyEvent(event);",
+                    "override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {",
+                    "TvRemoteHandlerModule.getInstance().onKeyEvent(event, \"up\");",
+                    "return super.onKeyUp(keyCode, event)",
+                    "}",
+                    "override fun onKeyLongPress(keyCode: Int, event: KeyEvent?): Boolean {",
+                    "   TvRemoteHandlerModule.getInstance().onKeyEvent(event,  \"longPress\");",
+                    "    return super.onKeyLongPress(keyCode, event)",
+                    "}",
+                    "override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {",
+                    "if(keyCode == KeyEvent.KEYCODE_DPAD_RIGHT || keyCode == KeyEvent.KEYCODE_DPAD_LEFT || keyCode == KeyEvent.KEYCODE_DPAD_UP || keyCode == KeyEvent.KEYCODE_DPAD_DOWN) {",
+                    "event?.startTracking();",
+                    "TvRemoteHandlerModule.getInstance().onKeyEvent(event,\"down\");",
+                    "return true;",
+                    "}",
+                    "TvRemoteHandlerModule.getInstance().onKeyEvent(event, \"down\");",
+                    "return super.onKeyDown(keyCode, event)",
                     "}"
                 ]
             }

